spec_summary/filters.go

```diff
@@ -0,0 +1,67 @@
+package spec_summary
+
+import "github.com/pkg/errors"
+
+type FilterKind = string
+type FilterValue = string
+type FilterValues = []FilterValue
+type Filters = map[FilterKind]FilterValues
```

Comment:
I used aliasing for clarity rather than enforcement, which I think is a modest improvement over base types.  If we want to go further, I'd rather do it in a separate PR.

Reply:
These types are exported. It will only be harder to change them later.

Reply:
Fair point.  I'll make `Filters`, `FilterKind`, and `FilterValues` proper types.  `FilterValue` really is string though, so I'll leave that as an alias.